frontend/src/components/metadata/DisplayMetadata.tsx

```diff
@@ -117,61 +118,76 @@ export const DisplayMetadata = (props: MetadataType) => {
-													initialReadOnly: true,
-													resourceId: resourceId,
-													content: metadata.content ?? null,
-													metadataId: metadata.id ?? null,
-													isRequired: field.required,
-													key: idxx,
-													datasetRole: datasetRole,
-												}
-											);
-										})
-									}
-									<Grid container spacing={2}>
-										<Grid item xs={11} sm={11} md={11} lg={11} xl={11}>
-											<Agent
-												created={metadata.created}
-												agent={metadata.agent}
-											/>
-											{datasetRole.role !== undefined &&
-											datasetRole.role !== "viewer" ? (
-												<MetadataDeleteButton
-													metadataId={metadata.id ?? null}
-													deleteMetadata={deleteMetadata}
-													resourceId={resourceId}
-													widgetName={metadataDef.name}
+				if (metadataList.length > 0) {
+					return currentMetadataDefList.map((metadataDef) => {
+						return metadataList.map((metadata, idx) => {
+							if (metadataDef.name === metadata.definition) {
+								return (
+									<Box className="inputGroup" key={idx}>
+										<Typography variant="h6">{metadata.definition}</Typography>
+										<Typography variant="subtitle2">
+											{metadata.description}
+										</Typography>
+										{
+											// construct metadata using its definition
+											metadataDef.fields.map((field, idxx) => {
+												return React.cloneElement(
+													metadataConfig[field.widgetType ?? "NA"] ??
+														metadataConfig["NA"],
+													{
+														widgetName: metadataDef.name,
+														fieldName: field.name,
+														options: field.config.options ?? [],
+														updateMetadata: updateMetadata,
+														initialReadOnly: true,
+														resourceId: resourceId,
+														content: metadata.content ?? null,
+														metadataId: metadata.id ?? null,
+														isRequired: field.required,
+														key: idxx,
+														datasetRole: datasetRole,
+													}
+												);
+											})
+										}
+										<Grid container spacing={2}>
+											<Grid item xs={11} sm={11} md={11} lg={11} xl={11}>
+												<Agent
+													created={metadata.created}
+													agent={metadata.agent}
 												/>
-											) : (
-												<></>
-											)}
+												{datasetRole.role !== undefined &&
+												datasetRole.role !== "viewer" ? (
+													<MetadataDeleteButton
+														metadataId={metadata.id ?? null}
+														deleteMetadata={deleteMetadata}
+														resourceId={resourceId}
+														widgetName={metadataDef.name}
+													/>
+												) : (
+													<></>
+												)}
+											</Grid>
 										</Grid>
-									</Grid>
-								</Box>
-							);
-						}
+									</Box>
+								);
+							}
+						});
 					});
-				});
+				} else {
+					return (
+						<Box textAlign="center">
```

Comment:
The center looks a bit weird. Could it be just align on the left? 
Thoughts? 
<img width="1455" alt="image" src="https://github.com/clowder-framework/clowder2/assets/13950475/7f99c341-3f11-4d2d-97e2-c1d97ba089e6">

Reply:
Aligned left.